tests/devices/qubit/test_adjoint_jacobian.py

```diff
@@ -0,0 +1,374 @@
+    def test_valid_tape_no_expand(self, G):
+        """Test that a tape that is valid doesn't raise errors and is not expanded"""
+        prep_op = qml.QubitStateVector(
+            np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0
+        )
+        qs = QuantumScript(
+            ops=[G(np.pi, wires=[0])], measurements=[qml.expval(qml.PauliZ(0))], prep=[prep_op]
+        )
+
+        qs.trainable_params = {1}
+        qs_valid = validate_and_expand_adjoint(qs)
+
+        assert all(qml.equal(o1, o2) for o1, o2 in zip(qs.operations, qs_valid.operations))
+        assert all(qml.equal(o1, o2) for o1, o2 in zip(qs.measurements, qs_valid.measurements))
+        assert qs.trainable_params == qs_valid.trainable_params
+
+    def test_valid_tape_with_expansion(self):
+        """Test that a tape that is valid with operations that need to be expanded doesn't raise errors
+        and is expanded"""
+        prep_op = qml.QubitStateVector(
+            np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0
+        )
+        qs = QuantumScript(
+            ops=[qml.Rot(0.1, 0.2, 0.3, wires=[0])],
+            measurements=[qml.expval(qml.PauliZ(0))],
+            prep=[prep_op],
+        )
+
+        qs.trainable_params = {1, 2, 3}
+        qs_valid = validate_and_expand_adjoint(qs)
+
+        expected_ops = [
+            prep_op,
+            qml.RZ(0.1, wires=[0]),
+            qml.RY(0.2, wires=[0]),
+            qml.RZ(0.3, wires=[0]),
+        ]
+
+        assert all(qml.equal(o1, o2) for o1, o2 in zip(qs_valid.operations, expected_ops))
+        assert all(qml.equal(o1, o2) for o1, o2 in zip(qs.measurements, qs_valid.measurements))
+        assert qs.trainable_params == qs_valid.trainable_params
+
+
+class TestAdjointJacobian:
+    """Tests for adjoint_jacobian"""
+
+    @pytest.fixture
+    def dev(self):
+        """Fixture that creates a two-qubit default qubit device for comparisions."""
+        return qml.device("default.qubit", wires=2)
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("theta", np.linspace(-2 * np.pi, 2 * np.pi, 7))
+    @pytest.mark.parametrize("G", [qml.RX, qml.RY, qml.RZ])
+    def test_pauli_rotation_gradient(self, G, theta, tol):
+        """Tests that the automatic gradients of Pauli rotations are correct."""
+
+        prep_op = qml.QubitStateVector(
+            np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0
+        )
+        qs = QuantumScript(
+            ops=[G(theta, wires=[0])], measurements=[qml.expval(qml.PauliZ(0))], prep=[prep_op]
+        )
+
+        qs.trainable_params = {1}
+        qs_valid = validate_and_expand_adjoint(qs)
+
+        calculated_val = adjoint_jacobian(qs_valid)
+
+        # compare to finite differences
+        dev_single_qubit = qml.device("default.qubit", wires=1)
+        tapes, fn = qml.gradients.finite_diff(qs)
+        numeric_val = fn(qml.execute(tapes, dev_single_qubit, None))
```

Comment:
```suggestion
        tapes, fn = qml.gradients.finite_diff(qs)
        results = tuple(qml.devices.qubit.simulate(t) for t in tapes)
        numeric_val = fn(results)
```

These tests should be independent of the old default qubit, as they should outlast them.